packages/components/tag/src/tag.vue

```diff
@@ -31,67 +31,41 @@
 <script lang="ts">
 import { computed, defineComponent } from 'vue'
 import { useGlobalConfig } from '@element-plus/utils/util'
-import { isValidComponentSize } from '@element-plus/utils/validators'
 
-import type { PropType } from 'vue'
-import type { ComponentSize } from '@element-plus/utils/types'
+import { tagProps, tagEmits } from './tag'
 
 export default defineComponent({
   name: 'ElTag',
-  props: {
-    closable: Boolean,
-    type: {
-      type: String as PropType<'success' | 'info' | 'warning' | 'danger' | ''>,
-      default: '',
-    },
-    hit: Boolean,
-    disableTransitions: Boolean,
-    color: {
-      type: String,
-      default: '',
-    },
-    size: {
-      type: String as PropType<ComponentSize>,
-      validator: isValidComponentSize,
-    },
-    effect: {
-      type: String,
-      default: 'light',
-      validator: (val: string): boolean => {
-        return ['dark', 'light', 'plain'].indexOf(val) !== -1
-      },
-    },
-  },
-  emits: ['close', 'click'],
-  setup(props, ctx) {
+
+  props: tagProps,
+  emits: tagEmits,
+
+  setup(props, { emit }) {
     const ELEMENT = useGlobalConfig()
 
-    const tagSize = computed(() => {
-      return props.size || ELEMENT.size
-    })
+    const tagSize = computed(() => props.size || ELEMENT.size)
     const classes = computed(() => {
       const { type, hit, effect } = props
       return [
         'el-tag',
         type ? `el-tag--${type}` : '',
         tagSize.value ? `el-tag--${tagSize.value}` : '',
         effect ? `el-tag--${effect}` : '',
-        hit && 'is-hit',
+        hit ? 'is-hit' : '',
```

Comment:
I think `hit && 'is-hit'` has no needs to change.

Reply:
`Vue` class combination will remove falsy(or non text type) class for us so that we don't need to do this.